pykeops/numpy/utils.py

```diff
@@ -119,6 +133,81 @@ def array(x, dtype=default_dtype, device=None):
     def device(x):
         return "cpu"
 
+    @staticmethod
+    def distance_function(metric):
+        def euclidean(x, y):
+            return ((x - y) ** 2).sum(-1)
+
+        def manhattan(x, y):
+            return ((x - y).abs()).sum(-1)
+
+        def angular(x, y):
+            return -(x | y)
+
+        def angular_full(x, y):
+            return angular(x, y) / ((angular(x, x) * angular(y, y)).sqrt())
+
+        if metric == "euclidean":
+            return euclidean
+        elif metric == "manhattan":
+            return manhattan
+        elif metric == "angular":
+            return angular
+        elif metric == "angular_full":
+            return angular_full
+        elif metric == "hyperbolic":
+            raise ValueError(
+                "Hyperbolic not supported for numpy, please use torch version with approximation"
+            )
+        else:
+            raise ValueError("Unknown metric")
```

Comment:
The error message could be a little bit clearer, e.g.: `f"Unknown metric: {metric}. Supported values are ..."`.

Reply:
done